internal/proxyservice/proxy_service.go

```diff
@@ -201,6 +215,38 @@ func (s *proxyServices) createSSHService(
 	return newSvc, nil
 }
 
+func (s *proxyServices) createSSHAgentService(
+	config v2.Service,
+	pluginInst tcp.Plugin,
+) (internal.Service, error) {
+
+	// TODO: Add validation somewhere about overlapping listenOns
+	// TODO: v2.NetworkAddress is a value type.  It needs to be moved to its
+	//   own package with no deps (stdlib deps are ok).
+	netAddr := v2.NetworkAddress(config.ListenOn)
+	listener, err := net.Listen(netAddr.Network(), netAddr.Address())
+	if err != nil {
+		return nil, err
+	}
+
+	connResources := s.connectorResources(config)
+	credsRetriever := s.credsRetriever(config.Credentials)
+
+	// TODO: NewProxyService needs to be injected
+	newSvc, err := sshagentproxy.NewProxyService(
+		listener,
+		connResources.Logger(),
+		credsRetriever,
+	)
+
+	if err != nil {
+		s.logger.Errorf("could not create proxy service '%s'", config.Name)
+		return nil, err
+	}
+
+	return newSvc, nil
+}
+
```

Comment:
https://github.com/cyberark/secretless-broker/pull/927#discussion_r334058255

Reply:
this is the stuff to be deferred to a later date?

Reply:
yup